addons/website_slides/models/slide_slide.py

```diff
@@ -58,9 +58,15 @@ def create(self, vals_list):
         return res
 
     def write(self, values):
+        slides_completed_change = self.env['slide.slide.partner']
+        if 'completed' in values:
+            slides_completed_change = self.filtered(lambda spr: spr.completed != values['completed'])
+
         res = super(SlidePartnerRelation, self).write(values)
-        if values.get('completed'):
-            self._set_completed_callback()
+
+        if slides_completed_change:
+            slides_completed_change._set_completed_callback()
```

Comment:
It's not really "set completed" anymore then.
How about renaming it simply to "_recompute_completion" to match what we have in slide.channel.partner?